src/lib/datatypes/include/sol-list.h

```diff
@@ -66,9 +66,15 @@ struct sol_list {
 
 /**
  * @def SOL_LIST_INIT
- * @brief Helper macro to initialize a @c sol_list structure.
+ * @brief Helper macro to initialize a @c sol_list structure. Example:
+ *
+ * @code
+ * struct sol_list mylist = SOL_LIST_INIT(mylist);
+ * @endcode
+ *
+ * @param name List to be initialized
  */
-#define SOL_LIST_INIT { NULL, NULL }
+#define SOL_LIST_INIT(name) { &(name), &(name) }
 
```

Comment:
@ceolin, I don't fully understand what you mean. As you can see in the test, the usage is the same as the code you provided. Following your example, you need a sentinel (list variable) to access the list API:

```c
struct sol_list delayed_packets = SOL_LIST_INIT(delayed_packets);

sol_list_add(&delayed_packets, &delayed_packate.list);
```

The sentinel must have ```next``` and ```prev``` pointers set to its address (that is the convention for an empty list). The limitation of ```SOL_LIST_INIT``` is that you have to pass a sol_list struct, and not a pointer to it, in the last case you have ```sol_list_init```. If you mean to define the macro as ```{ (name), (name) }```, then you'll have to have something like ```*my_list = SOL_LIST_INIT(my_list);``` which looks strange.